### PR TITLE
Initial update for 2.0.

### PR DESCRIPTION
**Pull request by cdata**

### CODE REVIEW

File: bower.json [hunk 1]
```diff
@@ -16,14 +16,14 @@
     "url": "git://github.com/PolymerElements/iron-meta.git"
   },
   "dependencies": {
-    "polymer": "Polymer/polymer#^1.0.0"
+    "polymer": "https://github.com/PolymerLabs/alacarte.git#master"
```
Comment (bicknellr) [inline bower.json]: `Polymer/polymer#2.0-preview`

File: iron-meta.html [hunk 2]
```diff
@@ -55,12 +55,43 @@
...
   (function() {
 
-    // monostate data
-    var metaDatas = {};
-    var metaArrays = {};
-    var singleton = null;
+    function IronMeta(options) {
+      this.type = (options && options.type) || 'default';
+      this.key = options && options.key;
+    }
```
Comment (bicknellr) [inline iron-meta.html]: This constructor should probably also handle the `value` property, if provided. `iron-iconset` and `iron-iconset-svg` have lines like this:
```javascript
new Polymer.IronMeta({type: 'iconset', key: this.name, value: this});
```
... just for the sake of putting itself into IronMeta's map. :/ It's easy to work around but I wouldn't be surprised to see it in other places.

**Comment by bicknellr** [comment]
BTW, `list` is no longer bindable because it's a getter.

**Comment by cdata** [comment]
Yup. Do we have any good concrete use cases for binding to `list`? I know
that it has been done, but `list` doesn't currently update if an
`iron-meta` in the distance changes its value. So, I'm not sure we're
really losing much here.

**Comment by bicknellr** [comment]
The only reason I bring it up is because iron-icons was using it in its demo. Getting around it was just a matter of manually pulling the list off of an `<iron-meta>` element (rather than `Polymer.IronMeta` instance, which doesn't have `list`) and manually setting it to the right property on a particular `dom-bind`. It might be useful to have an event or something to let you know that a new value has been put into a particular 'type' but I couldn't tell you if it's a common thing to care about or not.

File: bower.json [hunk 3]
```diff
@@ -16,14 +16,14 @@
     "url": "git://github.com/PolymerElements/iron-meta.git"
   },
   "dependencies": {
-    "polymer": "Polymer/polymer#^1.0.0"
+    "polymer": "polymer/polymer#2.0-preview"
```
Comment (bicknellr) [inline bower.json]: Thanks!

**Comment by cdata** [comment]
It's really useful, and something like `iron-meta` but based on
`AppStorageBehavior` would probably do the trick.

`iron-meta`'s practical usage is rigidly defined by its cost to create
(very high compared to just looking up a property on an object). So, we are
deprecating it. In `v1.x.x`, it's a footgun to suggest that binding to
`list` is safe. It's not, because `list` won't update over time. Since we
are deprecating the element, I think we should avoid enhancing it any
further.

**Comment by bicknellr** [comment]
> Since we are deprecating the element, I think we should avoid enhancing it any further.

sounds good

File: iron-meta.html [hunk 4]
```diff
@@ -268,66 +137,64 @@
...
-                break;
-            }
-          }
+      __computeMeta: function(type, key, value) {
+        let meta = new Polymer.IronMeta({
+          type: type,
+          key: key
+        });
+
+        if (value !== undefined && value !== meta.value) {
```
Comment (e111077) [inline iron-meta.html]: I'm going to wait on reviewing this element as setting `value`as a setter and getter seems to be acting weird with Polymer despite it being set on the element's markdown.

File: iron-meta.html [hunk 5]
```diff
@@ -268,66 +137,64 @@
...
           type: Object,
           notify: true,
-          readOnly: true
         },
 
         /**
-         * Array of all meta-data values for the given type.
+         * If true, `value` is set to the iron-meta instance itself.
          */
-        list: {
```
Comment (bicknellr) [inline iron-meta.html]: Would it be possible to make this a computed property rather than a getter?

Comment (bicknellr) [inline iron-meta.html]: Oh, we discussed this already.

**Comment by notwaldorf** [comment]
@e111077 undo your red x, we're calling this one ready :)

**Pull request merged**